### PR TITLE
Fix descriptor sets free validation error

### PR DESCRIPTION
**Pull request by rukai**
Seeing as kvark believes this change to be correct I've opened this PR.
Still doesnt fix https://github.com/gfx-rs/wgpu/issues/221 though

### CODE REVIEW

File: wgpu-native/src/command/compute.rs [hunk 1]
```diff
@@ -66,7 +66,11 @@ pub extern "C" fn wgpu_compute_pass_set_bind_group(
     let mut pass_guard = HUB.compute_passes.write();
     let pass = &mut pass_guard[pass_id];
     let bind_group_guard = HUB.bind_groups.read();
-    let bind_group = &bind_group_guard[bind_group_id];
+
+    let bind_group = pass
+        .trackers
+        .bind_groups
+        .get_with_extended_usage(&*bind_group_guard, bind_group_id, DummyUsage).unwrap();
```
Comment (kvark) [inline wgpu-native/src/command/compute.rs]: actually, I think this is done in the `insert_barriers` below, so this particular change shouldn't be needed

**Comment by rukai** [comment]
Looks like this PR is superseded by https://github.com/gfx-rs/wgpu/pull/226
Or at the very least I should wait until https://github.com/gfx-rs/wgpu/pull/226 lands before continuing work with this.

**Comment by kvark** [comment]
It would still be needed. Ideally, we'd be figuring out what the problem is while #226 is maturing, and then add a proper fix on top.

**Comment by kvark** [comment]
@rukai looks like we are missing the submission index updates near the "// update submission IDs" comment. Could you add bind groups there and double check if it solves the issue?

**Comment by kvark** [comment]
There is also a merge missing inside `insert_barriers`...

**Comment by rukai** [comment]
Thanks! This PR now fixes https://github.com/gfx-rs/wgpu/issues/221

**Pull request merged**